cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -34,20 +34,34 @@ function(find_libarrow_in_python_wheel PYARROW_VERSION)
   # version number soname, just `${MAJOR_VERSION}00`
   set(PYARROW_LIB "libarrow.so.${PYARROW_SO_VER}00")
 
-  find_package(Python REQUIRED)
   execute_process(
```

Comment:
Why do we need this execute_process here? This looks the same as the initial_code_block that will be called on line 67 once we find Arrow?